lib/ui/widgets/components/picker_item.dart

```diff
@@ -36,25 +36,30 @@ class PickerItem<T extends Object> {
 
 // TODO(reddwarf03): specify [PickerItem.value] types (thanks to Generics) (3)
 class PickerWidget<T extends Object> extends ConsumerStatefulWidget {
-  const PickerWidget({
+  PickerWidget({
     super.key,
     this.pickerItems,
     this.onSelected,
-    this.selectedIndex = -1,
-    this.showSelectedItem = true,
-  });
+    this.onUnselected,
+    this.selectedIndexes,
+    this.multipleSelectionsAllowed = false,
+  }) {
+    // the list cannot have a const value because we will always need (even for mono selection cases) to add/remove elements within it
+    if (this.selectedIndexes == null) {
+      this.selectedIndexes = [];
+    }
+  }
   final ValueChanged<PickerItem<T>>? onSelected;
+  final ValueChanged<PickerItem<T>>? onUnselected;
   final List<PickerItem<T>>? pickerItems;
-  final int selectedIndex;
-  final bool showSelectedItem;
+  late List<int>? selectedIndexes;
```

Comment:
? is not necessary if you put `late` keyword i think

Reply:
I will take it the other way, I will delete the `late` keyword, I need the `?` keyword.